### PR TITLE
Refactoring to add a Flow Decomposition Result builder

### PR DESCRIPTION
**Pull request by OpenSuze**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowDecompositionComputer.java [hunk 1]
```diff
@@ -162,94 +149,64 @@ private LoadFlowRunningService.Result runDcLoadFlow(Network network) {
...
     }
 
-    private void saveDcReferenceFlow(FlowDecompositionResults flowDecompositionResults, List<Branch> xnecList) {
-        flowDecompositionResults.saveDcReferenceFlow(getXnecReferenceFlows(xnecList));
+    private void saveDcReferenceFlow(FlowDecompositionResultsBuilder flowDecompositionResultsBuilder, List<Branch> xnecList) {
+        flowDecompositionResultsBuilder.saveDcReferenceFlow(getXnecReferenceFlows(xnecList));
     }
 
-    private SensitivityAnalyser getSensitivityAnalyser(Network network, NetworkMatrixIndexes networkMatrixIndexes) {
+    SensitivityAnalyser getSensitivityAnalyser(Network network, NetworkMatrixIndexes networkMatrixIndexes) {
```
Comment (murgeyseb) [inline flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowDecompositionComputer.java:179]: I understand that you need to make it private package to allow testing, but can't you find another solution that avoid breaking encapsulation ?

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/AllocatedFlowTests.java [hunk 2]
```diff
@@ -23,119 +21,40 @@
...
-        String allocated = "Allocated Flow";
 
-        Network network = importNetwork(networkFileName);
-        FlowDecompositionParameters flowDecompositionParameters = new FlowDecompositionParameters();
-        flowDecompositionParameters.setSaveIntermediates(FlowDecompositionParameters.SAVE_INTERMEDIATES);
-        FlowDecompositionComputer allocatedFlowComputer = new FlowDecompositionComputer(flowDecompositionParameters);
-        FlowDecompositionResults flowDecompositionResults = allocatedFlowComputer.run(network);
+        Network network = TestUtils.importNetwork(networkFileName);
+        FlowDecompositionComputer allocatedFlowComputer = new FlowDecompositionComputer();
+        FlowDecompositionResults flowDecompositionResultsBuilder = allocatedFlowComputer.run(network);
```
Comment (murgeyseb) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/AllocatedFlowTests.java:31]: flowDecompositionResults (without builder)

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/AllocatedFlowTests.java [hunk 3]
```diff
@@ -23,119 +21,40 @@
...
```
Comment (murgeyseb) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/AllocatedFlowTests.java:55]: flowDecompositionResults (without builder)

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/InternalFlowTests.java [hunk 4]
```diff
@@ -29,52 +29,51 @@ class InternalFlowTests {
 
     @Test
     void testNetworkWithoutInternalFlow() {
-        Network network = AllocatedFlowTests.importNetwork(NETWORK_FILE_NAME);
+        Network network = TestUtils.importNetwork(NETWORK_FILE_NAME);
 
-        FlowDecompositionResults flowDecompositionResults = getFlowDecompositionResults(network);
+        FlowDecompositionResults flowDecompositionResultsBuilder = getFlowDecompositionResults(network);
```
Comment (murgeyseb) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/InternalFlowTests.java:34]: flowDecompositionResults (without builder)

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/InternalFlowTests.java [hunk 5]
```diff
@@ -29,52 +29,51 @@ class InternalFlowTests {
...
+        assertEquals(Country.FR, flowDecompositionResultsBuilder.getDecomposedFlowMap().get(X_LFR_LBE).getCountry1());
+        assertEquals(Country.BE, flowDecompositionResultsBuilder.getDecomposedFlowMap().get(X_LFR_LBE).getCountry2());
     }
 
     @Test
     void testNetworkWithInternalFlow() {
         Network network = getHighPtdfNetwork();
 
-        FlowDecompositionResults flowDecompositionResults = getFlowDecompositionResults(network);
+        FlowDecompositionResults flowDecompositionResultsBuilder = getFlowDecompositionResults(network);
```
Comment (murgeyseb) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/InternalFlowTests.java:49]: flowDecompositionResults (without builder)

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/PstFlowTests.java [hunk 6]
```diff
@@ -24,51 +23,29 @@ class PstFlowTests {
...
-        FlowDecompositionResults flowDecompositionResults = flowComputer.run(network);
-
-        var optionalPsdfs = flowDecompositionResults.getPsdfMap();
-        assertTrue(optionalPsdfs.isPresent());
-        var psdf = optionalPsdfs.get();
-        assertEquals(-420.042573, psdf.get(x1).get(pst), EPSILON);
-        assertEquals(420.042573, psdf.get(x2).get(pst), EPSILON);
+        Network network = TestUtils.importNetwork(networkFileName);
+        FlowDecompositionComputer flowComputer = new FlowDecompositionComputer();
+        FlowDecompositionResults flowDecompositionResultsBuilder = flowComputer.run(network);
```
Comment (murgeyseb) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/PstFlowTests.java:31]: flowDecompositionResults (without builder)

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/PstFlowTests.java [hunk 7]
```diff
@@ -24,51 +23,29 @@ class PstFlowTests {
...
-        FlowDecompositionResults flowDecompositionResults = flowComputer.run(network);
-
-        var optionalPsdfs = flowDecompositionResults.getPsdfMap();
-        assertTrue(optionalPsdfs.isPresent());
-        var psdf = optionalPsdfs.get();
-        assertEquals(-420.042573, psdf.get(x1).get(pst), EPSILON);
-        assertEquals(-420.042573, psdf.get(x2).get(pst), EPSILON);
+        Network network = TestUtils.importNetwork(networkFileName);
+        FlowDecompositionComputer flowComputer = new FlowDecompositionComputer();
+        FlowDecompositionResults flowDecompositionResultsBuilder = flowComputer.run(network);
```
Comment (murgeyseb) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/PstFlowTests.java:46]: flowDecompositionResults (without builder)

**Review by murgeyseb — CHANGES_REQUESTED** [review]

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/AllocatedFlowTests.java (hunk 2 above)
Comment (OpenSuze) [reply flow-decomposition/src/test/java/com/powsybl/flow_decomposition/AllocatedFlowTests.java:31]: corrected

**Review by murgeyseb — APPROVED** [review]

**Pull request merged**